app/controllers/charges_controller.rb

```diff
@@ -194,6 +194,13 @@ def create_square_payment_request(
 
     # Creates a pending PaymentIntent. See webhooks_controller to see what
     # happens when the PaymentIntent is successful.
+    
+    # Check for whether a campaign exits. If a campaign exists, then check whether the campaign has a non-profit associated with it. 
+    # If a non-profit exists, then check whether the non-profit has a fee ID. If so, then set fee ID equal to fee_id
+    fee_id = if @campaign && @campaign[:nonprofit_id] && Nonprofit.find_by(id: @campaign[:nonprofit_id])[:fee_id]
```

Comment:
If you don't mind, do you mind writing a test for this in the charges_controller_spec?